File: 53-maximum-subarray/maximum-subarray.go
func maxSubArray(nums []int) int {
	currNum := nums[0]
	maxSum := nums[0]
	for i := 1; i < len(nums); i++ {
		currNum = max(nums[i], currNum+nums[i])
		maxSum = max(maxSum, currNum)
	}

	return maxSum
}

File: 53-maximum-subarray/maximum-subarray.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var maxSubArray = function (nums) {

    maxSum = nums[0]
    currNum = nums[0]

    for (let i = 1; i < nums.length; i++) {
        currNum = Math.max(currNum + nums[i] , nums[i])
        maxSum = Math.max(maxSum, nums[i], currNum)
    }
    return maxSum
};